qiime2/plugin/plugin.py

```diff
@@ -67,9 +69,10 @@ def __init__(self, name, version, website, package, citation_text=None,
 
         self.formats = {}
         self.views = {}
-        self.types = {}
+        self.type_fragments = {}
         self.transformers = {}
         self.type_formats = []
+        self.types = set()
```

Comment:
So this becomes an `@property` producing a dictionary of `str(semantic_type)` to `SemanticTypeRecord`?  Because that's what `get_semantic_types` does, so should we just make that an `@property`?